olp-cpp-sdk-dataservice-read/src/StreamLayerClientImpl.cpp

```diff
@@ -378,6 +378,65 @@ client::CancellableFuture<PollResponse> StreamLayerClientImpl::Poll() {
                                                         std::move(promise));
 }
 
+client::CancellationToken StreamLayerClientImpl::Seek(
+    SeekRequest request, SeekResponseCallback callback) {
+  auto seek_task = [=](client::CancellationContext context) -> SeekResponse {
+    std::string subscription_id;
+    std::string subscription_mode;
+    std::string x_correlation_id;
+    std::shared_ptr<client::OlpClient> client;
+
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      if (!client_context_) {
+        OLP_SDK_LOG_WARNING_F(kLogTag, "Seek: unsuccessful, not subscribed");
+        return client::ApiError(client::ErrorCode::PreconditionFailed,
+                                "Not subscribed", false);
```

Comment:
```suggestion
                                "Subscription missing", false);
```

Maybe better.

Reply:
Unsubscribe and Poll methods have same error and message. Should I change them all to "Subscription missing"?

Reply:
Yes, please.